pkg/controller/kogitodataindex/resource/statefulset.go

```diff
@@ -26,14 +26,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 	"path"
+	"strconv"
 )
 
 func newStatefulset(instance *v1alpha1.KogitoDataIndex, secret *corev1.Secret, externalURI string, cli *client.Client) (*appsv1.StatefulSet, error) {
+	// define the http port
+	httpPort := defineDataIndexHTTPPort(instance)
+	log.Debugf("The configured internal Data Index port number is [%i]", httpPort)
+
 	// create a standard probe
 	probe := defaultProbe
-	probe.Handler.TCPSocket = &corev1.TCPSocketAction{Port: intstr.FromInt(defaultExposedPort)}
+	probe.Handler.TCPSocket = &corev1.TCPSocketAction{Port: intstr.IntOrString{IntVal: httpPort}}
 	// environment variables
 	removeManagedEnvVars(instance)
+
+	// set KOGITO_DATA_INDEX_HTTP_PORT env
+	hasEnv := false
+	for envName := range instance.Spec.Env {
+		if envName == DataIndexEnvKeyHTTPPort {
+			hasEnv = true
+		}
+	}
+	// if there is no env or there's no dataIndex http port env, add it, otherwise just override its value
+	if len(instance.Spec.Env) == 0 || !hasEnv {
+		envMap := make(map[string]string)
+		envMap[DataIndexEnvKeyHTTPPort] = strconv.FormatInt(int64(httpPort), 10)
+		log.Debugf("Has no %s Env, adding it to spec %s", DataIndexEnvKeyHTTPPort, envMap)
+		instance.Spec.Env = util.AppendStringMap(instance.Spec.Env, envMap)
+	} else if hasEnv {
+		log.Debugf("Env [%s] found, overriding its value  from [%i] to [%i]",
+			DataIndexEnvKeyHTTPPort, instance.Spec.Env[DataIndexEnvKeyHTTPPort], strconv.FormatInt(int64(httpPort), 10))
+		instance.Spec.Env[DataIndexEnvKeyHTTPPort] = strconv.FormatInt(int64(httpPort), 10)
+	}
+
 	// from cr
 	envs := instance.Spec.Env
 	envs = util.AppendStringMap(envs, fromKafkaToStringMap(externalURI))
```

Comment:
I think we should add the new port environment variable to `envs` here, instead of modifying the envs of the data index CR.

Reply:
I did it this way to make sure that, even if, for some reason, user define the env on the cr, we can identify and override it accordingly. Does it make sense?

Reply:
Oh wait, I can override it, I can replace all this logic with only one line.

Reply:
updated, thanks;